src/process/message_processor.go

```diff
@@ -1,48 +1,94 @@
 package process
 
 import (
+	"BGPAlert/analyze"
 	"BGPAlert/common"
 	"errors"
 	"fmt"
 	"time"
 )
 
-// Constantly reads channel of messages and stores them in Window objects to send through windowChannel for analysis
-func ProcessBGPMessages(msgChannel chan common.BGPMessage, windowChannel chan common.Window) error {
-	var bucketMap = make(map[time.Time][]common.BGPMessage)
+// Constantly read messages from channel, build up windows with frequency maps, calling analysis on windows when full
+func ProcessBGPMessages(msgChannel chan common.BGPMessage) error {
+	maximumBuckets := 43
+	// Each bucket is 60s
+	parseDuration, err := time.ParseDuration("60s")
+	if err != nil {
+		return errors.New("error creating parsing bucket window, " + err.Error())
+	}
+	maximumTimespan := time.Duration(maximumBuckets*60) * time.Second
+
+	window := common.Window{
+		Filter:    "none",
+		BucketMap: make(map[time.Time][]common.BGPMessage),
+	}
 
+	// Read messages from channel
 	for msg := range msgChannel {
-		windowSize, err := time.ParseDuration("60s")
-		if err != nil {
-			return errors.New("error creating parsing window, " + err.Error())
-		}
+		// Round timestamp to size of parseDuration to place in correct bucket
+		messageBucket := msg.Timestamp.Truncate(parseDuration)
+
+		// If a bucket with this timestamp doesn't exist, need to create one
+		if _, ok := window.BucketMap[messageBucket]; !ok {
+			fmt.Println("----------------------------------------------------------------------------------")
+
+			// If there's at least one timestamp, may have to fill in missing zeroes in map
+			if len(window.BucketMap) > 1 {
+				// Get minimum timestamp from map
+				minTimestamp := getBucketMapMin(window.BucketMap)
```

Comment:
trying to think through the logic here, and when this *really* needs to be triggered: I think it's maybe most important when you create a new bucket? i.e., if you have to create a new bucket, perhaps there's a gap you need to fill. But if you don't have to create a new bucket, then no change?